hack/jenkins/e2e-runner.sh

```diff
@@ -383,7 +411,12 @@ fi
 # * neither started nor destroyed (soak test)
 if [[ "${E2E_UP:-}" == "${E2E_DOWN:-}" && -f "${gcp_resources_before}" && -f "${gcp_resources_after}" ]]; then
   difference=$(diff -sw -U0 -F'^\[.*\]$' "${gcp_resources_before}" "${gcp_resources_after}") || true
+  noleak=true
   if [[ -n $(echo "${difference}" | tail -n +3 | grep -E "^\+") ]] && [[ "${FAIL_ON_GCP_RESOURCE_LEAK:-}" == "true" ]]; then
+    noleak=false
+  fi
+  record_command "${STAGE_CLEANUP}" "resource-leak" ${noleak}
+  if ! ${noleak} ; then
```

Comment:
This is pretty confusing. At the least switch `noleak` to `leak` and get rid of this double negation, and I'd really rather explicitly check `[[ "${leak}" == "true" ]]`.